python/nix_csi/service.py

```
@@ -70,24 +73,41 @@ async def NodePublishVolume(self, stream):
+                            )
+                            extraArgs = [
+                                "--extra-substituters",
+                                "ssh-ng://nix@nix-cache?trusted=1&priority=20",
+                            ]
+                            logger.debug("Cache connectivity check succeeded")
+                            break
+                        except (GRPCError, OSError, asyncio.TimeoutError) as e:
+                            if attempt < 2:
+                                logger.debug(f"Cache check attempt {attempt + 1}/3 failed, retrying: {e}")
+                            else:
+                                logger.warning(f"Cache unavailable after 3 attempts, building without cache")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)